src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -85,6 +85,13 @@ private void MainContextCanceledEvent(object state)
             }
         }
 
+        private void RelayCurrentContextLogsToLogger(object sender, LoggingEventArgs e)
+        {
+            // Relay logs that are generated by the current MLContext to the Experiment class's
+            // _logger.
+            _logger.Info(e.Message);
```

Comment:
We should be able the maintain the message kind:
```suggestion
            switch (e.Kind)
            {
                case ChannelMessageKind.Trace:
                    _logger.Trace(e.Message);
                     break;
                case ChannelMessageKind.Info:
                    _logger.Info(e.Message);
                     break;
                case ChannelMessageKind.Warning:
                    _logger.Warning(e.Message);
                     break;
                case ChannelMessageKind.Error:
                    _logger.Error(e.Message);
                     break;
                 default:
                     throw new NotImplementedException($"{nameof(ChannelMessageKind)}.{e.Kind.ToString()} is not yet implemented.");
            }
```

I wonder if there's a way to just pass `e` along directly without accessing e.Message ([code](https://github.com/dotnet/machinelearning/blob/609003edc3254f69b85feeee933e6c833e6f3f95/src/Microsoft.ML.Data/LoggingEventArgs.cs#L48-L51))?

Reply:
This is very neat, thank you! `LoggingEventArgs` doesn't have a default `ToString()` unfortunately, hence we'll have to use `e.Message`.